logstash-core/src/test/resources/org/logstash/config/ir/cache/pipeline_reuse_baseline.conf

```diff
@@ -17,6 +17,7 @@ filter {
   if [a] {
     if [a] {
       if [a] {
+        noop {}
```

Comment:
### Note for reviewer
This PR solves the problem to protect against if failures inserting a try-catch statement in the dataset that invokes the split dataset that evaluates the `EventCondition`.

without this PR the following statement:

```
if [a] {
    noop "hit"
}
```

is translated to the following Dataset graph:
```mermaid
graph TD;
DatasetFC[Dataset filter call]-->Split[Split Dataset];
    DatasetFC-->FiltDlg[Filter Delegate.multiFilter 'hit'];
    Split-->EC[Event condition];
    Split-->CMPL[Complement];
```

and the following filters call-chain reuse the same `Dataset filter call`. So
```
  mutate "hit"
  mutate "hit 2"
```
is translated to the following AST
```mermaid
graph TD;
DatasetFC[Dataset filter call]-->DatasetFC2[Dataset filter call];
    DatasetFC-->FiltDlg[Filter Delegate.multiFilter 'hit2'];
    DatasetFC2-->CMPL[Next...];
    DatasetFC2-->FiltDlg2[Filter Delegate.multiFilter 'hit'];
```

while with new change it becomes:
```mermaid
graph TD;
DatasetFC[Dataset filter call]-->Split[Split Dataset];
    DatasetFC-->FiltDlg[Filter Delegate.multiFilter 'hit'];
    Split-->EC[Event condition];
    DatasetFC-->ErrList[error handler]
    Split-->CMPL[Complement];
```

So while in without this PR the same Dataset filter was used both inside an `if` statement and to chain filter calls. With this PR the dataset that wraps a filter after an `if` statement are different and so the reason why to create this "noop chain" which is used also in the `pipeline_reuse_test.conf`, so if we don't chain here we don't have the dataset that wraps the filter without the try statement.